src/audio/dai-zephyr.c

```diff
@@ -635,12 +612,12 @@ static int dai_playback_params(struct dai_data *dd, struct comp_dev *dev, uint32
 					&max_block_count);
 		if (err < 0) {
 			comp_err(dev, "dai_playback_params(): could not get dma attr max block count, err = %d", err);
-			goto out;
+			return err;
 		}
 
 		if (!max_block_count) {
 			comp_err(dev, "dai_playback_params(): invalid max-block-count of zero");
-			goto out;
+			return err;
```

Comment:
Logic remains the same as before, no idea why it was like this though.

Reply:
Lets keep the same logic and update in another PR if logic can be improved.

Reply:
@lgirdwood the logic **is** preserved here. I was just commenting on an inconsistency, whereby an error message is printed but 0 is returned, signalling success. Yes, by all means this can be addressed separately